src/coprocessor/circom.rs

```diff
@@ -0,0 +1,115 @@
+//! # Usage of circom coprocessors.
+//!
+//! See `examples/circom.rs` for a quick example of how to declare a circom coprocessor.
+
+use core::fmt::Debug;
+use std::path::PathBuf;
+
+use anyhow::Result;
+use bellperson::{ConstraintSystem, SynthesisError};
+use nova_scotia::r1cs::CircomConfig;
+
+use crate::{
+    circuit::gadgets::{
+        circom::CircomGadget,
+        data::GlobalAllocations,
+        pointer::{AllocatedContPtr, AllocatedPtr},
+    },
+    coprocessor::{CoCircuit, Coprocessor},
+    field::LurkField,
+    ptr::Ptr,
+    store::Store,
+};
+
+/// TODO: fix this duplication with the one in `cli::paths`
+fn circom_gadgets() -> PathBuf {
```

Comment:
How would do you recommend we populate the circom binary in CI? Have it installed at the host? OK, but where?
Can I pass its location as an environment variable?

Reply:
Yes, I guess there should be an environment variable for this, then the host will have to install it